earth_enterprise/src/common/geFilePoolImpl.h

```diff
@@ -20,18 +20,34 @@
 
 #include <khMTTypes.h>
 #include <fcntl.h>
+#include <string>
 
 class geFilePool;
 
+class AbstractFileIdentifier {
+public:
+  virtual bool isValid() { return false; }
+  virtual int getAsFD() { return -1; }
+  virtual void invalidate() {};
+};
+
+class POSIXIdentifier: public AbstractFileIdentifier {
+  int fd;
+public:
+  POSIXIdentifier(int i = -1) : fd{ i } {}
+  bool isValid() override { return fd != -1; }
+  void invalidate() override { fd = -1; }
+  int getAsFD() override { return fd; }
+};
 
 class FileReservationImpl : public khMTRefCounter {
   bool isWriter;
-  int fd;
+  AbstractFileIdentifier* fid;
```

Comment:
In coordination with the `shared_ptr` changes suggested above, this should be `std::shared_ptr<AbstractFileIdentifier> fid;`